### PR TITLE
Turn single radio button into text

### LINKED ISSUE #3810: Burotel: display only "Daily booking" text instead of radio button
Since in burotel this is the only option, it's pretty useless to show a single "radio button" for it.

Happens in:
  * Landing;
  * When creating booking from List of Spaces;
  * when editing a booking.

### PR DESCRIPTION
**Pull request by panagiotappl**
closes #3810

### CODE REVIEW

File: indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx [hunk 1]
```diff
@@ -207,36 +207,41 @@ class BookingEditForm extends React.Component {
         const bookedByCurrentUser = sessionUser.id === bookedForUser.id;
         const today = moment();
         const bookingFinished = today.isAfter(startDt, 'day') && (endDt && today.isAfter(endDt, 'day'));
-
+        const onlyDaily = hideOptions.single && hideOptions.recurring && !hideOptions.daily;
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: shouldn't this check for all possible combinations where only one option is visible?

Comment (panagiotappl) [reply indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: In the code for the Landing page the logic is a bit different and we only check for `onylDaily`, so I wasn't sure what kind of approach I should follow.

Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: I guess we should change it there as well then :)

Comment (panagiotappl) [reply indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: Discard the prop `onlyDaily` and replace it with `hideOptions.single` etc?

**Comment by ThiefMaster** [comment]
i wouldn't mention burotel in the commit message since all this commit does is hiding the radio buttons if there's only a single option selected - the fact that the burotel plugin uses this doesn't matter for this commit

File: indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx (hunk 1 above)
Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: yep

**Comment by ThiefMaster** [comment]
image


I don't think this looks amazing... what about hiding this text altogether when there's only one option?

File: indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx [hunk 2]
```diff
@@ -208,35 +208,40 @@ class BookingEditForm extends React.Component {
         const today = moment();
         const bookingFinished = today.isAfter(startDt, 'day') && (endDt && today.isAfter(endDt, 'day'));
 
+        const onlySingle = !hideOptions.single && hideOptions.recurring && hideOptions.daily;
+        const onlyDaily = hideOptions.single && hideOptions.recurring && !hideOptions.daily;
+        const onlyRecurring = hideOptions.single && !hideOptions.recurring && hideOptions.daily;
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx]: You could simplify these checks like this:

```js
// all but one option are hidden
const showRecurrenceOptions = ['single', 'daily', 'recurring'].filter(x => hideOptions[x]).length !== 2;
```

and then just use `showRecurrenceOptions` below

**Pull request merged**